React/component-design.md

```diff
@@ -0,0 +1,40 @@
+# Component Design
+
+## Learning Goals
+- Examine design patterns used by React developers
+- Use the functional stateless components design pattern
+- Use the container components with presentational components pattern
+
+## Design Patterns
+In software engineering, a [design pattern](https://en.wikipedia.org/wiki/Software_design_pattern) is a generic, reusable solution to a commonly occurring problem. In this discussion, we will examine two design patterns in React.
+
+## Functional Stateless Components
+The name of this type of component provides some insight into the functionality and behavior of the component.
+
+The two key features of these components are:
+- No lifecycle methods, only rendering
+- Can utilize `props` but not `state`
+
+These are used to simplify a component when it's only purpose is to render HTML or other nested components. The definition of these components use a simplified arrow-function syntax since they don't need to be a fully functional `class` like other components.
+
+```JavaScript
+const MyFunctionalComponent = (props) => {
+  return <h1>This is just a header with this name: {props.name}</h1>;
+}
+```
+
+Notice in the above example we can still pass in props into this component, but we access that variable directly from the parameter passed in, rather than through `this.props` like we would with a class-based React component.
```

Comment:
Can you have them immediately apply this to the livecode? Maybe the `<Student>` component, unless some other yet-to-be-written lecture makes it stateful.